src/components/tabs/tabs.ts

```diff
@@ -0,0 +1,391 @@
+            this.headersContent.offsetWidth -
+            this.offset
+        ) {
+          this.scrollElement(tab, true);
+        }
+
+        if (
+          tab.offsetLeft + tab.offsetWidth >
+          this.headersWrapper.offsetWidth - this.offset
+        ) {
+          this.scrollElement(tab, false);
+        }
+      }
+    } else {
+      this.hideSelectedIndicator();
+    }
+  }
+
+  private handleClick(event: MouseEvent) {
+    const target = event.target as HTMLElement;
+    const tab = target.closest('igc-tab');
+
+    if (!tab || tab.disabled) {
+      return;
+    }
+
+    tab.focus();
+    this.setSelectedTab(tab.panel);
+  }
+
+  private handleKeyDown = (event: KeyboardEvent) => {
+    const { key } = event;
+    const activeEl = document.activeElement;
+    const enabledTabs = this.tabs.filter((el) => !el.disabled);
+
+    let index = enabledTabs.indexOf(activeEl as IgcTabComponent);
+
+    switch (key) {
+      case 'ArrowLeft':
+        index = this.isLTR
+          ? (enabledTabs.length + index - 1) % enabledTabs.length
+          : (index + 1) % enabledTabs.length;
+        break;
+      case 'ArrowRight':
+        index = this.isLTR
+          ? (index + 1) % enabledTabs.length
+          : (enabledTabs.length + index - 1) % enabledTabs.length;
+        break;
+      case 'Home':
+        index = 0;
+        break;
+      case 'End':
+        index = enabledTabs.length - 1;
+        break;
+      case 'Enter':
+      case ' ':
+        this.setSelectedTab(enabledTabs[index].panel);
+        break;
+      default:
+        return;
+    }
+
+    enabledTabs[index].focus({ preventScroll: true });
+
+    if (this.activation === 'auto') {
+      this.setSelectedTab(enabledTabs[index].panel);
+    }
+
+    this.scrollTabIntoView(enabledTabs[index]);
+
+    event.preventDefault();
+  };
+
+  protected override render() {
+    return html`
+      <div part="headers">
+        ${this.showStartScrollButton
+          ? html`
+              <igc-icon-button
+                part="start-scroll-button"
+                size="medium"
+                variant="flat"
+                name="navigate_before"
+                collection="internal"
+                @click=${this.handleStartButtonClick}
+              ></igc-icon-button>
+            `
+          : ''}
+        <div part="headers-content">
+          <div part="headers-wrapper">
+            <div part="headers-scroll" role="tablist">
+              <slot></slot>
+            </div>
+            <div part="selected-indicator"></div>
+          </div>
+        </div>
+        ${this.showEndScrollButton
+          ? html`
+              <igc-icon-button
+                part="end-scroll-button"
+                size="medium"
+                variant="flat"
+                name="navigate_next"
+                collection="internal"
+                @click=${this.handleEndButtonClick}
+              ></igc-icon-button>
+            `
+          : ''}
```

Comment:
Consider extracting the scroll button rendering into a separate renderer which takes the condition and the props.
Then this could be reduced to:
```html
<div part="headers">
    ${this.renderScrollButton(condition, {...props})}
    <div part="headers-content">
        <div part="headers-wrapper">
          <div part="headers-scroll" role="tablist">
            <slot></slot>
          </div>
          <div part="selected-indicator"></div>
        </div>
    </div>
    <!-- OR ${this.renderHeaders() } -->
    ${this.renderScrollButton(condition, {...props})}
</div>
```